pkg/tsdb/influxdb/flux/builder.go

```diff
@@ -175,14 +183,44 @@ func getTimeSeriesTimeColumn(columns []*query.FluxColumn) *query.FluxColumn {
 	return nil
 }
 
+func getTableID(record *query.FluxRecord, groupColumns []string) []interface{} {
+	result := make([]interface{}, len(groupColumns))
+
+	for i, colName := range groupColumns {
+		result[i] = record.ValueByKey(colName)
```

Comment:
So I was thinking what will happen here if we have duplicate columns. If there are 2 `table` columns here we still can access only one which would be a bit weird imho. Looking at the docs it seems like table change should be handled by `if result.TableChanged() {` https://github.com/influxdata/influxdb-client-go#querytableresult. And we seem to be doing that here https://github.com/grafana/grafana/blob/e0448513eb29bd9f5f435e4585de3c34a56dee72/pkg/tsdb/influxdb/flux/executor.go#L56. So then in https://github.com/grafana/grafana/blob/cfaca42e973fa86efb62f10ed01586fbd582953a/pkg/tsdb/influxdb/flux/builder.go#L185 as code stands right this is basically an initialisation (because init sets table to -1 and that first append will do this) and probably isn't designed to handle table changes in between appends.

Obviously the `table` column thrown this off but that does not seem like a good way to handle tables anyway.

Personally I think we could try to throw away the code checking table at all inside the builder and just say that this handles single table and you should make sure you use it correctly and create new builder `if result.TableChanged()` (optionally just return error if that went wrong). I think that would simplify this quite a bit. We could then just check if we are doing append for the first time and if so, just run single init code.

Reply:
unfortunately flux seems to call two different things "table", both what one might call "series" and a grouping of series that has the same schema, and `TableChanged` is only called for when the second thing changes, and not when the first :-(

regarding the duplicate names, i checked and `flux`, except the table-thing,  does not allow duplicate column-names. will update the code with a comment about this.